fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -476,11 +476,11 @@ protected static void addCacheControlHeaders(final HttpServletResponse servletRe
 
         final FedoraResource mutableResource = resource instanceof NonRdfSourceDescription
                 ? ((NonRdfSourceDescription) resource).getDescribedResource() : resource;
-        final EntityTag etag = new EntityTag(mutableResource.getEtagValue());
+        final EntityTag etag = new EntityTag(mutableResource.getEtagValue(), true);
         final Date date = mutableResource.getLastModifiedDate();
 
         if (!etag.getValue().isEmpty()) {
-            servletResponse.addHeader("ETag", etag.toString());
+            servletResponse.addHeader("ETag", mutableResource.hasType(FEDORA_NON_RDF_SOURCE_DESCRIPTION) ? "" : "W/" ) + etag);
```

Comment:
I just reread your comment (https://github.com/fcrepo4/fcrepo4/pull/920#issuecomment-151646374) and discovered that I had stashed some changes when I last worked onthis.  Will sort this out in a bit.